client-misk/src/main/kotlin/app/cash/backfila/client/misk/EmbeddedBackfillRun.kt

```diff
@@ -0,0 +1,203 @@
+package app.cash.backfila.client.misk
+
+import app.cash.backfila.protos.clientservice.GetNextBatchRangeRequest
+import app.cash.backfila.protos.clientservice.GetNextBatchRangeResponse
+import app.cash.backfila.protos.clientservice.KeyRange
+import app.cash.backfila.protos.clientservice.PrepareBackfillRequest
+import app.cash.backfila.protos.clientservice.PrepareBackfillResponse
+import app.cash.backfila.protos.clientservice.RunBatchRequest
+import okio.ByteString
+import okio.ByteString.Companion.encodeUtf8
+import java.util.ArrayDeque
+
+//TODO(mikepaw) do I need to do anything to make this more explicitly not thread safe?
```

Comment:
is this class only used for tests? then no need for thread safe. I wonder if we want a `client-misk-test` module so you don't always pull in the testing code

Reply:
Yeah I guess I could extract it out. Might be a good idea.

Reply:
The comment has been updated